src/rgw/rgw_sal_motr.cc

```diff
@@ -1481,12 +1472,18 @@ int MotrObject::MotrReadOp::read(int64_t off, int64_t end, bufferlist& bl, optio
 int MotrObject::MotrReadOp::iterate(const DoutPrefixProvider* dpp, int64_t off, int64_t end, RGWGetDataCB* cb, optional_yield y)
 {
   int rc;
-
-  if (source->category == RGWObjCategory::MultiMeta)
+  // Open and read the object here.
+  if (source->category == RGWObjCategory::MultiMeta){
+    ldpp_dout(dpp, 20) <<__func__<< ": open obj parts..." << dendl;
+    source->get_part_objs(dpp, this->part_objs)? : source->open_part_objs(dpp, this->part_objs);
```

Comment:
check return status and return if opening object failing in both cases.

Reply:
done. please check